webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/getFileValueStandard.scala.txt

```diff
@@ -36,39 +36,22 @@ PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
 PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
 PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
 
-SELECT ?fileValue ?objPred ?objObj
-WHERE {
-
+CONSTRUCT {
+    ?fileValue ?objPred ?objObj .
+    ?fileValue knora-base:attachedToProject ?resourceProject .
+    ?fileValue knora-base:hasPermissions ?currentFileValuePermissions .
+} WHERE {
     ?fileValue knora-base:internalFilename "@filename" .
 
     ?currentFileValue knora-base:previousValue* ?fileValue ;
-        knora-base:isDeleted false .
-
-    ?prop rdfs:subPropertyOf* knora-base:hasFileValue .
+        knora-base:hasPermissions ?currentFileValuePermissions .
```

Comment:
I am not sure I remember why this is necessary (although I seem to have worked on that [query](https://github.com/dasch-swiss/knora-api/commit/edc953d85a30b00db45c3450869290719899a030#diff-e6e71faf6639e7e8512da4ebfa4f29fd)): I suppose it's necessary because permissions are only stored for the current version of a value, and if someone asks for a filename that was changed with a later version of the file value, it's the only way to get the permissions, right?

So this means permissions are not versioned.

Reply:
> The permissions attached to the current version of a value also apply to previous versions of the value. Value versions other than the current one do not have this predicate.

https://docs.knora.org/02-knora-ontologies/knora-base/#permissions

Reply:
I just realised that this means that the `FILTER` is redundant: 05029e9f

Reply:
(Values don't have `attachedToProject` anymore, either.)

Reply:
> I just realised that this means that the `FILTER` is redundant: [05029e9](https://github.com/dasch-swiss/knora-api/commit/05029e9f965938327e1d32b8681a068fe9c16223)

true, good catch!